recommenders/models/rbm/rbm.py

```diff
@@ -738,17 +638,17 @@ def fit(self, xtr, xtst):
             if self.with_metrics and i % self.display == 0:
                 log.info("training epoch %i rmse %f" % (i, epoch_tr_err))
 
-            Rmse_train.append(epoch_tr_err)  # mse training error per training epoch
+            rmse_train.append(epoch_tr_err)  # mse training error per training epoch
 
         # optionally evaluate precision metrics
         precision_train, precision_test = self.train_test_precision(xtst)
-        elapsed = self.time()
 
-        log.info("done training, Training time %f2" % elapsed)
+        # stop the timer
+        self.timer.stop()
```

Comment:
the timer here wraps all the fit function, this should be outside. See for example how we do it https://github.com/microsoft/recommenders/blob/main/examples/00_quick_start/sar_movielens.ipynb
One important principle we follow is single responsibility: https://github.com/Microsoft/Recommenders/wiki/Coding-Guidelines#single-responsibility each function does only one thing. In this case, the fit function only trains, it doesn't compute the time, it doesn't test. If we follow this principle, we have a much more modular code. Please review the other timer functions in the code. It only makes sense to have the timer inside the code if we are doing some iterative loging, otherwise, we can take it out and simplify the code

Reply:
Removed timer from the model class. Execution times are measured separately in the notebooks now.